OpenRA.Game/Game.cs

```diff
@@ -457,6 +465,28 @@ static string ChooseShellmap()
 			return shellmaps.Random(CosmeticRandom);
 		}
 
+		public static void SwitchToExternalMod(ExternalMod mod, string[] launchArguments = null, Action onFailed = null)
+		{
+			try
+			{
+				var argsString = mod.LaunchArgs.Append(launchArguments)
+					.Select(a => "\"" + a + "\"").JoinWith(" ");
+
+				var p = Process.Start(mod.LaunchPath, argsString);
```

Comment:
I think this has severe security implications. You are letting the user start new processes from within the game and plan to do so from online sources in the future. That will essentially become a malware distribution service.